service/src/main/java/org/cbioportal/service/util/ProfiledCasesCounter.java

```diff
@@ -86,15 +86,7 @@ public void calculate(List<String> molecularProfileIds, List<String> sampleIds,
                 // add alterationCount object where there are no alterations but have genePanel
                 // object
                 if (!genesWithAlteration.containsKey(entrezGeneId)) {
-                    AlterationCountByGene alterationCountByGene = null;
-
-                    if (alterationCounts.get(0) instanceof MutationCountByGene) {
```

Comment:
This would throw error `alterationCounts` can be empty (when there are no alterations for queried cases). Instead directly create an object for `AlterationCountByGene`